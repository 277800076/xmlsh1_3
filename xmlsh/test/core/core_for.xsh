# core_for.xsh
# tests of for

for a in a b c ; do
	echo $a ;
done

for a in a b c ; do echo $a ; done

# test leftover variable
echo After for a is $a

set xa xb xc
for a ; do echo $a ; done

# test IO redirection

[ -d $TMPDIR/_xmlsh ] || mkdir $TMPDIR/_xmlsh
T1=$TMPDIR/_xmlsh/for1.tmp
T2=$TMPDIR/_xmlsh/for2.tmp

echo line1 > $T1
echo line2 >> $T1
echo line3 >> $T1

for a in a b c  ; do
   read line 
   echo $line
done < $T1 > $T2

echo Output is 
cat $T2

rm $T1 $T2 
