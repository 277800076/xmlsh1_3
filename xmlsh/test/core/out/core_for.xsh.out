a
b
c
a
b
c
After for a is c
xa
xb
xc
Output is
line1
line2
line3
